Modules/CTP/src/CTPTrendingTask.cxx

```diff
@@ -41,17 +41,23 @@ void CTPTrendingTask::configure(const boost::property_tree::ptree& config)
 {
   mConfig = TrendingConfigCTP(getID(), config);
 }
-
-void CTPTrendingTask::initialize(Trigger t, framework::ServiceRegistryRef services)
+void CTPTrendingTask::initCTP(Trigger& t)
 {
-  // // read out the CTPConfiguration
   std::string run = std::to_string(t.activity.mId);
-  CTPRunManager::setCCDBHost("https://alice-ccdb.cern.ch");
-  mCTPconfig = CTPRunManager::getConfigFromCCDB(t.timestamp, run);
-
+  // CTPRunManager::setCCDBHost("https://alice-ccdb.cern.ch");
+  // mCTPconfig = CTPRunManager::getConfigFromCCDB(t.timestamp, run);
+  std::string CCDBHost = "https://alice-ccdb.cern.ch";
+  auto& mgr = o2::ccdb::BasicCCDBManager::instance();
+  mgr.setURL(CCDBHost);
+  map<string, string> metadata; // can be empty
+  metadata["runNumber"] = run;
+  mCTPconfig = mgr.getSpecific<CTPConfiguration>(CCDBPathCTPConfig, t.timestamp, metadata);
+  if (mCTPconfig == nullptr) {
+    return;
+  }
   // get the indeces of the classes we want to trend
-  std::vector<ctp::CTPClass> ctpcls = mCTPconfig.getCTPClasses();
-  std::vector<int> clslist = mCTPconfig.getTriggerClassList();
+  std::vector<ctp::CTPClass> ctpcls = mCTPconfig->getCTPClasses();
+  std::vector<int> clslist = mCTPconfig->getTriggerClassList();
```

Comment:
In the loop below, you use `clslist.size()` but then you access `ctpcls[i]`. Is `clslist` really needed? Wouldn't be better to use `ctpcls.size()` in the loop?

Reply:
@aferrero2707 thanks, we shall check, cc @lhusova .